File: Source/Geometry/outer_product.js
function outer_product(v1, v2) {
    let result = [];
    for (let i = 0; i < v1.length; i++) {
        let row = [];
        for (let j = 0; j < v2.length; j++) {
            row.push(v1[i] * v2[j]);
        }
        result.push(row);
    }
    return result;
}

File: Source/Geometry/outer_product.c
#include <stdlib.h>

double** outer_product(double v1[], double v2[], int n) {
    double** result = (double**)malloc(n * sizeof(double*));
    for (int i = 0; i < n; i++) {
        result[i] = (double*)malloc(n * sizeof(double));
        for (int j = 0; j < n; j++) {
            result[i][j] = v1[i] * v2[j];
        }
    }
    return result;
}
